deployer/src/main/java/brooklyn/entity/cloudfoundry/CloudFoundryEntity.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package brooklyn.entity.cloudfoundry;
+
+import brooklyn.config.ConfigKey;
+import brooklyn.entity.Entity;
+import brooklyn.entity.basic.Attributes;
+import brooklyn.entity.basic.BrooklynConfigKeys;
+import brooklyn.entity.basic.ConfigKeys;
+import brooklyn.entity.basic.Lifecycle;
+import brooklyn.entity.proxying.ImplementedBy;
+import brooklyn.entity.trait.Startable;
+import brooklyn.event.AttributeSensor;
+import brooklyn.event.basic.Sensors;
+import brooklyn.location.cloudfoundry.CloudFoundryPaasLocation;
+import brooklyn.util.flags.SetFromFlag;
+import brooklyn.util.time.Duration;
+
+@ImplementedBy(CloudFoundryEntityImpl.class)
+public interface CloudFoundryEntity extends Entity, Startable {
```

Comment:
I suspect this new entity hierarchy is not really needed, isn't it?

I think we agreed on trying to have a design that reuse the same entity definition and using the driver (the only changing part) the deployer will be able to deploy on IaaS or PaaS?

I appreciate `newDriver` requires a `MachineLocation` which you don't have as `PaasLocation` doesn't extend it, maybe we should patch brooklyn code to accept a more generic AbstractLocation instead, rather than adding this new duplicate hierarchy of entities. wdyt?